src/mame/drivers/galaxian_rockclim.cpp

```diff
@@ -0,0 +1,294 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+
+#include "emu.h"
+#include "includes/galaxian.h"
+
+class galaxian_rockclim_state : public galaxian_state
+{
+public:
+	galaxian_rockclim_state(const machine_config &mconfig, device_type type, const char *tag)
+		: galaxian_state(mconfig, type, tag)
+		, m_rockclim_videoram(*this,"rockclim_vram")
+		, m_gfxdecode2(*this,"gfxdecode2")
+		, m_palette2(*this,"palette2")
+	{ }
+
+	void rockclim(machine_config &config);
+
+	void init_rockclim();
+
+private:
+	virtual void video_start() override;
+
+	void rockclim_map(address_map &map);
+
+	void rockclim_extend_sprite_info(const uint8_t *base, uint8_t *sx, uint8_t *sy, uint8_t *flipx, uint8_t *flipy, uint16_t *code, uint8_t *color);
+	void rockclim_videoram_w(offs_t offset, uint8_t data);
+	void rockclim_scroll_w(offs_t offset, uint8_t data);
+	uint8_t rockclim_videoram_r(offs_t offset);
+	optional_shared_ptr<uint8_t> m_rockclim_videoram;
```

Comment:
Why is this optional?  You’ve only got one machine configuration on this variant of the platform.  Also, please keep private data members together, and private member functions together.

Reply:
it's optional as it was optional before, just needs a bit of tidying up